iterative_refinement/expectation_maximization.py

```diff
@@ -0,0 +1,306 @@
+"""
+Iterative refinement in Bayesian expection maximization setting
+for reconstruction of particles.
+"""
+
+import numpy as np
+from compSPI.transforms import do_fft, do_ifft
+from simSPI.transfer import eval_ctf
+
+
+class IterativeRefinement:
+    """
+    Performs interative refimenent in a Bayesian expectation
+    maximization setting, i.e. maximum a posteriori estimation.
```

Comment:
Hmmm, it is MAP or just MLE?

There is no prior, so we're doing maximum likelihood, not maximum a posteriori.

For MAP, I think there needs to be something like a prior over the shift, ctf, or rotation. The only search here is over rotation, and there is no prior on it... besides an implicit uniform (over SO(3)) prior.

Reply:
i changed the docstring to emphasize mle